### ISSUE TITLE
Multiline JSDoc Types

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#16179]
From https://github.com/Microsoft/vscode/issues/27777

**Issue**
For the javascript:

```js
/**
 * @param {{
 *  z: number
 * }} x
 */
function foo(x){}
```

The type of `x` is `any`. It seems multiline types are not recognized in jsdocs

There seems to be some debate about whether this is supported or not: https://github.com/jscs-dev/node-jscs/issues/245 but it seems closure and some other tools support the multiline syntax

Also tracking syntax highlighting for this case here: https://github.com/Microsoft/TypeScript-TmLanguage/issues/467

**Comment by seanf** [comment]
As a workaround, it may help to delete the leading `*` characters on the lines in between `{{` and `}}`. Assuming your linter will let you.

**Comment by rob2d** [comment]
@seanf did not seem to help in my use-cases; would be great if this were built in without needing an explicit linter

**Comment by sandersn** [comment]
@rob2d can you post an example of the code you couldn't get to work? I tried @mjbvz's example, and after deleting the leading asterisks, the type works:

```js
/**
 * @param {{
   z: number
  }} x
 */
function foo(x){  x.z }
```

This is a known bug #23667

**Comment by weswigham** [comment]
AFAIK this is fixed now.

**Comment by nomaed** [comment]
@weswigham is it?

With VSCode Version 1.30.1:

![screen shot 2019-01-04 at 11 54 12](https://user-images.githubusercontent.com/9551921/50682429-7e411f00-1017-11e9-82a7-6e76060dca43.png)

**Comment by weswigham** [comment]
AFAIK the syntax highlighting is still waiting on https://github.com/atom/language-javascript/pull/515

**Comment by nomaed** [comment]
Yes, I found this link a bit after asking about it. Thanks!

**Comment by RodrigoTomeES** [comment]
hi, @weswigham Atom was abandoned, Microsoft will implement the syntax highlight?

Also I found an issue with the multi-line JSDoc but I don't know if this will be here or I need to create another issue because it's related with multi-line JSDoc Types.

The bug is that if you use multi-line the type is not inferred but if I put all in one line the type is inferred.

https://github.com/microsoft/TypeScript/assets/13077343/2bfb845f-062c-496c-a88b-6274f9d6c7fd

**Comment by sandersn** [comment]
Please do file a new bug. This one tracked the addition of the feature, not bugs with it.

When you do, can you provide the text of your program or a link to the typescript playground with your program included?

**Comment by loganmzz** [comment]
@sandersn I don't understand your remark, the question is about to support multiline declaration of types and it just doesn't work... So current issue is just not 'done' 🤔

**Comment by RyanCavanaugh** [comment]
Please do file a new bug
